### PR TITLE
Fix service delete event

### PR DESCRIPTION
**Pull request by zmingxi**
**What type of PR is this?**
bug

**Which issue does this PR fix**:
https://github.com/aws/aws-application-networking-k8s/issues/237

**What does this PR do / Why do we need it**:
Fix

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:
make e2etest
Manually run https://github.com/aws/aws-application-networking-k8s/issues/257

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No
**Does this PR introduce any user-facing change?**:
No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: controllers/eventhandlers/endpoints.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (h *enqueueRequestsForEndpointsEvent) enqueueImpactedService(queue workqueu
 	}
 
 	if err := h.client.Get(context.TODO(), namespaceName, svc); err != nil {
-		glog.V(2).Infof("enqueueImpactedService, service not found %v\n", err)
+		glog.V(2).Infof("Event: enqueueImpactedService, service not found %v\n", err)
```
Comment (liwenwu-amazon) [inline controllers/eventhandlers/endpoints.go:79]: let's use V(6)

File: controllers/eventhandlers/service.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ func NewEqueueRequestServiceEvent(client client.Client) handler.EventHandler {
 
 func (h *enqueueRequetsForServiceEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
 	service := e.Object.(*corev1.Service)
+	h.enqueueImpactedService(queue, service)
```
Comment (liwenwu-amazon) [inline controllers/eventhandlers/service.go:32]: being consistent , can we add glog.V(6) here

File: controllers/eventhandlers/service.go [hunk 3]
```diff
@@ -37,13 +38,34 @@ func (h *enqueueRequetsForServiceEvent) Update(e event.UpdateEvent, queue workqu
 
 func (h *enqueueRequetsForServiceEvent) Delete(e event.DeleteEvent, queue workqueue.RateLimitingInterface) {
 	service := e.Object.(*corev1.Service)
+	h.enqueueImpactedService(queue, service)
```
Comment (liwenwu-amazon) [inline controllers/eventhandlers/service.go:41]: same as above

File: controllers/service_controller.go [hunk 4]
```diff
@@ -101,23 +102,29 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 
 	if err := r.Client.Get(ctx, req.NamespacedName, svc); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	if !svc.DeletionTimestamp.IsZero() {
+		tgNameD := latticestore.TargetGroupName(svc.Name, svc.Namespace)
+		TGDeleted := ds.GetTargetGroupsByTG(tgNameD)
+		for _, tg := range TGDeleted {
+			glog.V(6).Infof("endpoints change trigger target IP list registration %v and tg %v\n",
```
Comment (liwenwu-amazon) [inline controllers/service_controller.go:115]: Please update this debugging log line

File: pkg/gateway/model_build_targets_test.go [hunk 5]
```diff
@@ -72,6 +81,38 @@ func Test_Targets(t *testing.T) {
 				},
 			},
 		},
+		{
```
Comment (liwenwu-amazon) [inline pkg/gateway/model_build_targets_test.go:84]: Can you add a unit test to test service delete ?

Comment (zmingxi) [reply pkg/gateway/model_build_targets_test.go:84]: Unit test is covered at line 85-115

File: pkg/gateway/model_build_targets_test.go [hunk 6]
```diff
@@ -72,6 +81,38 @@ func Test_Targets(t *testing.T) {
...
+			name:               "Add all endpoints to build spec",
```
Comment (liwenwu-amazon) [inline pkg/gateway/model_build_targets_test.go:85]: Please update this unit test to match what this unit test for?

File: pkg/gateway/model_build_targets_test.go (hunk 5 above)
Comment (liwenwu-amazon) [reply pkg/gateway/model_build_targets_test.go:84]: also, can u add a new test for just service add without HTTProute stuff...

Comment (zmingxi) [reply pkg/gateway/model_build_targets_test.go:84]: Created issue for adding e2etest: https://github.com/aws/aws-application-networking-k8s/issues/257

**Comment by liwenwu-amazon** [comment]
LGTM.  Please also add more detail on how you fixed the issue. Thanks

**Review by liwenwu-amazon — APPROVED** [review]

**Pull request merged**